contracts/gateway/Gateway.sol

```diff
@@ -0,0 +1,202 @@
+     *  deployments requested with a valid signature from an approved signer will be accepted.
+     *  If the gateway does not require a signature,
+     *  @param signer the signer address to approve.
+     */
+    function approveSigner(address signer) external onlyOwner {
+        if (signer == address(0)) {
+            revert ZeroAddress();
+        }
+
+        if (approvedSigners[signer]) {
+            revert SignerAlreadyApproved();
+        }
+
+        approvedSigners[signer] = true;
+
+        emit SignerApproved(signer);
+    }
+
+    /**
+     *  @dev Revoke a signer to sign ONCHAINID deployments.
+     *  @param signer the signer address to revoke.
+     */
+    function revokeSigner(address signer) external onlyOwner {
+        if (signer == address(0)) {
+            revert ZeroAddress();
+        }
+
+        if (!approvedSigners[signer]) {
+            revert SignerAlreadyNotApproved();
+        }
+
+        delete approvedSigners[signer];
+
+        emit SignerRevoked(signer);
+    }
+
+    /**
+     *  @dev Deploy an ONCHAINID using a factory. The operation must be signed by
+     *  an approved public key. This method allow to deploy an ONCHAINID using a custom salt.
+     *  @param identityOwner the address to set as a management key.
+     *  @param salt to use for the deployment.
+     *  @param signatureExpiry the block timestamp where the signature will expire.
+     *  @param signature the approval containing the salt and the identityOwner address.
+     */
+    function deployIdentityWithSalt(
+        address identityOwner,
+        string memory salt,
+        uint256 signatureExpiry,
+        bytes calldata signature
+    ) external returns (address) {
+        if (identityOwner == address(0)) {
+            revert ZeroAddress();
+        }
+
+        if (signatureExpiry != 0 && signatureExpiry < block.timestamp) {
+            revert ExpiredSignature();
+        }
+
+        address signer = ECDSA.recover(
+            keccak256(
+                abi.encode(
+                    "Authorize ONCHAINID deployment",
+                    identityOwner,
+                    salt,
+                    signatureExpiry
+                )
+            ).toEthSignedMessageHash(),
+            signature
+        );
+
+        if (!approvedSigners[signer]) {
+            revert UnapprovedSigner();
+        }
+
+        if (revokedSignatures[signature]) {
+            revert RevokedSignature();
+        }
+
+        return idFactory.createIdentity(identityOwner, salt);
+    }
+
+    /**
+     *  @dev Deploy an ONCHAINID using a factory using the identityOwner address as salt.
+     *  @param identityOwner the address to set as a management key.
+     */
+    function deployIdentityForWallet(address identityOwner) external returns (address) {
+        if (identityOwner == address(0)) {
+            revert ZeroAddress();
+        }
+        if (identityOwner != msg.sender) {
```

Comment:
do we really need to force users to deploy their own identity? i believe we could deploy identities on behalf of users, as in any case you cannot steal identities this way. the identity ownership (management key) will be given to the wallet of `identityOwner`

Reply:
Mh, let's remove this requirement. => https://github.com/onchain-id/solidity/pull/89/commits/202868f6885c3ced60d11fed4991d160c38f0d37